### ISSUE TITLE
Runtime error in intrinsic "execute_command_line" when the "cmdstat" argument is missing

### ISSUE THREAD

**Issue by psteinfeld** [llvm/llvm-project#77979]
When I call the intrinsic "execute_command_line" and do not pass an actual argument for "cmdstat", the program gets a fatal runtime error.  Here's a program that shows the problem:
```
program bug
   implicit none
   integer                       :: exitstatvar, cmdstatvar
   character(len=256)            :: msg
   character(len=:), allocatable :: command
   command='notthere'
   msg=''
   exitstatvar = 216
   cmdstatvar = 343
   msg = ''
   cmdstatvar = 3
   call execute_command_line(command , wait=.false., exitstat=exitstatvar, cmdmsg=msg )
   print *, "test 2"
   print *, "cmdstat: ", cmdstatvar
   print *, "exitstat: ", exitstatvar
   print *, "msg: ", msg
end program bug
```
When I compile and execute this program, I get the output:
```
 test 2
 cmdstat:  3
 exitstat:  216
 msg:                                                                          
                                                                               
                                                                               
                            
[psteinfeld@ice4 install]$ sh: notthere: command not found

fatal Fortran runtime error(/local/home/psteinfeld/test/install/nocmdstat.f90:12): Execution error with system status code: -1
```
This does not happen when the `exitstat` or the `cmdmsg` argument is missing.

**Comment by yi-wu-arm** [comment]
This is actually how it should behave as standard described in 16.9.73
"If a condition occurs that would assign a nonzero value to `CMDSTAT` but the `CMDSTAT` variable is not present, error termination is initiated."

**Comment by yi-wu-arm** [comment]
However the execution error with system return -1 is wrong, the issue is list here: https://github.com/llvm/llvm-project/issues/77803 and will be fixed in here: https://github.com/llvm/llvm-project/pull/77944

**Comment by psteinfeld** [comment]
> However the execution error with system return -1 is wrong, the issue is list here: #77803 and will be fixed in here: #77944

Thanks!  I didn't notice that.  I'm closing this.
